### PR TITLE
Refactor all to use GetX reactive variables

### PR DESCRIPTION
**Pull request by VASANTH04032004**
Refactor to use GetX reactive variables and state management

- Replaced all `var` instances with reactive types (`RxString`, `RxBool`) across the codebase.
- Updated methods and logic to use GetX’s reactive state management for improved responsiveness and clean code.
- Refactored all `StatelessWidget` instances to `GetView` for automatic controller injection and cleaner code.
- Refined loading state handling with reactive variables.
- Enhanced error handling and success feedback using GetX snackbar notifications.

### CODE REVIEW

File: lib/screens/otp_verification_screen.dart [hunk 1]
```diff
@@ -4,7 +4,7 @@ import 'package:get/get.dart';
 import '../controllers/auth_controller.dart';
 
 class OTPVerificationScreen extends StatelessWidget {
```
Comment (AnaghaVB) [inline lib/screens/otp_verification_screen.dart]: Change this to a GetView

File: lib/screens/otp_verification_screen.dart [hunk 2]
```diff
@@ -4,7 +4,7 @@ import 'package:get/get.dart';
...
-  final AuthController _authController = Get.find();
+  final AuthController authController = Get.find();
```
Comment (AnaghaVB) [inline lib/screens/otp_verification_screen.dart]: Do not need this if you pass the AuthController to GetView

Comment (VASANTH04032004) [reply lib/screens/otp_verification_screen.dart]: ok

File: lib/screens/otp_verification_screen.dart (hunk 1 above)
Comment (VASANTH04032004) [reply lib/screens/otp_verification_screen.dart]: sorry not notice  that.

**Review by AnaghaVB — APPROVED** [review]

**Pull request merged**